src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -4701,6 +4724,12 @@ CachedRelationNamespaceLookupExtended(const char *relationName, Oid relnamespace
 {
 	/* force callbacks to be registered, so we always get notified upon changes */
 	InitializeCaches();
+	Oid citusExtensionOid = get_extension_oid("citus", true);
+
+	if (creating_extension && CurrentExtensionObject == citusExtensionOid)
+	{
+		CachedDuringCitusCreation = GetCurrentTransactionId();
```

Comment:
We can use `GetCurrentTransactionIdIfAny()` instead but we shouldn't call AssignTransactionId if we were to use GetCurrentTransaction() since we're in the middle of "Create Extension" which should be a valid transaction. Also, thanks for the catch on GetCurrentTransaction() not returning an Oid.

Reply:
`creating_extension` is only true when doing `execute_extension_script()`, which is always called after some other catalog work, so it will have an transaction ID already. I guess that's not guaranteed, but it also doesn't seem like wasted effort because any extension creation will need to get a transaction ID at some point.

That being said, maybe `GetCurrentTransactionNestLevel()` might be a better fit. We just need to know enough to know when to clear the cache.

Reply:
> creating_extension is only true when doing execute_extension_script(), which is always called after some other catalog work, so it will have an transaction ID already. I guess that's not guaranteed, but it also doesn't seem like wasted effort because any extension creation will need to get a transaction ID at some point.

Worth adding that context in a comment.